lib/synapse/config_generator/haproxy.rb

```diff
@@ -1153,13 +1158,37 @@ def generate_backend_stanza(watcher, config)
               b = "#{b} cookie #{backend_name}"
             end
           end
-          b = "#{b} #{watcher_config['server_options']}" if watcher_config['server_options'].is_a? String
-          b = "#{b} #{backend['haproxy_server_options']}" if backend['haproxy_server_options'].is_a? String
+
+          if @opts['use_nerve_weights'] && backend['weight'] && ((backend['weight'].is_a? String) || (backend['weight'].is_a? Integer))
```

Comment:
why the 'weight' field can be either a string or a integer? should it just be a integer?

Reply:
or, if string input is really a valid case, should we check whether it represents a valid integer? otherwise, I think ruby will just infer the value as 0 on #1174.

Reply:
I changed this to support only integers.  An invalid string would weight to 0, which could unexpectedly disable a backend.